functional-tests/common/com.liferay.ide.functional.swtbot/src/com/liferay/ide/functional/swtbot/eclipse/page/MavenPreferencesDialog.java

```diff
@@ -0,0 +1,119 @@
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.ide.functional.swtbot.eclipse.page;
+
+import com.liferay.ide.functional.swtbot.page.Button;
+import com.liferay.ide.functional.swtbot.page.CLable;
+import com.liferay.ide.functional.swtbot.page.CheckBox;
+import com.liferay.ide.functional.swtbot.page.Dialog;
+import com.liferay.ide.functional.swtbot.page.Text;
+import com.liferay.ide.functional.swtbot.page.TextInGroup;
+
+import org.eclipse.swtbot.swt.finder.SWTBot;
+
+/**
+ * @author Rui Wang
+ */
+public class MavenPreferencesDialog extends Dialog {
+
+	public MavenPreferencesDialog(SWTBot bot) {
+		super(bot);
+	}
+
+	public Button applyBtn() {
+		return new Button(bot, APPLY);
+	}
+
+	public CheckBox getAddPluginTypeSuffixForMavenProjectContextRoot() {
+		return new CheckBox(getShell().bot(), ADD_PLUGIN_TYPE_SUFFIX_FOR_MAVEN_PROJECT_CONTEXT_ROOT);
+	}
+
+	public TextInGroup getDefaultArchetypes() {
+		return new TextInGroup(bot, DEFAULT_ARCHETYPES_FOR_NEW_LIFERAY_PLUGIN_PROJECT_WIZARD, 0);
+	}
+
+	public CheckBox getDisableCustomJspValidationChecking() {
+		return new CheckBox(getShell().bot(), DISABLE_CUSTOM_JSP_VALIDATION_CHECKING);
+	}
+
+	public Text getExt() {
+		return new Text(getShell().bot(), EXT);
+	}
+
+	public Text getHook() {
+		return new Text(getShell().bot(), HOOK);
+	}
+
+	public Text getLayoutTemplate() {
+		return new Text(getShell().bot(), LAYOUT_TEMPLATE_PREFERENCE);
+	}
+
+	public TextInGroup getMavenProjectConfiguratorOption() {
+		return new TextInGroup(bot, MAVEN_PROJECT_CONFIGURATOR_OPTIONS, 1);
+	}
+
+	public Text getPortlet() {
+		return new Text(getShell().bot(), PORTLET_PREFERENCE);
+	}
+
+	public Text getPortletICEfaces() {
+		return new Text(getShell().bot(), PORTLET_ICEFACES);
+	}
+
+	public Text getPortletJsf() {
+		return new Text(getShell().bot(), PORTLET_JSF);
+	}
+
+	public Text getPortletLiferayFaceAlloy() {
+		return new Text(getShell().bot(), PORTLET_LIFERAY_FACES_ALLOY);
+	}
+
+	public Text getPortletPrimefaces() {
+		return new Text(getShell().bot(), PORTLET_PRIMEFACES);
+	}
+
+	public Text getPortletRichfaces() {
+		return new Text(getShell().bot(), PORTLET_RICHFACES);
+	}
+
+	public Text getPortletSpringMVC() {
+		return new Text(getShell().bot(), PORTLET_SPRING_MVC);
+	}
+
+	public Text getPortletVaadin() {
+		return new Text(getShell().bot(), PORTLET_VAADIN);
+	}
+
+	public Text getServiceBuilder() {
+		return new Text(getShell().bot(), SERVICE_BUILDER_PERERENCE);
+	}
+
+	public Text getTheme() {
+		return new Text(getShell().bot(), THEME_PREFERENCE);
+	}
+
+	public Text getWeb() {
+		return new Text(getShell().bot(), WEB);
+	}
+
+	public CLable mavenLable() {
+		return new CLable(bot, MAVEN);
+	}
+
+	public Button restoreDefaultsBtn() {
```

Comment:
Every page has "Apply" and "Resotre Default", we should not add them to every files. Try to move them and every page can call them.